### PR TITLE
Consider 'Incomplete' as a marker

### PR DESCRIPTION
**Pull request by asanin-epfl**
Solution to #323

### CODE REVIEW

**Comment by arnaudon** [comment]
Wow, thanks for that!

File: src/readers/morphologyASC.cpp [hunk 1]
```diff
@@ -281,6 +281,12 @@ class NeurolucidaParser
                 }
                 return true;
             } else if (is_end_of_branch(id)) {
```
Comment (tomdele) [inline src/readers/morphologyASC.cpp:283]: I am really bad with the asc format.  Are we sure the `Incomplete` is always detected as a end of branch ? We can never have something like : 

```
( -266.79  -131.77   -26.13     
Incomplete       
```
or even:
```
 ( -266.79  -131.77   -26.13     0.92) 
 Incomplete
 ( -268.17  -130.62   -24.75     0.92)
```
before a end of split ?

Comment (asanin-epfl) [reply src/readers/morphologyASC.cpp:283]: I understand that 'Incomplete' always ends the branch from my discussion with @arnaudon. We can additionally validate other cases and throw.

Comment (arnaudon) [reply src/readers/morphologyASC.cpp:283]: Yes, it is to mark a branch in incomplete, so it must be at a leave branch. I think you can safely throw if not

**Review by tomdele — APPROVED** [review]
Looks Great ! 
Thanks Aleksei

**Comment by asanin-epfl** [comment]
Thank you!

**Pull request merged**